nixos/modules/config/nix.nix

```diff
@@ -332,6 +332,16 @@ in
                 always allowed to connect.
               '';
             };
+
+            flake-registry = mkOption {
+              type = types.str;
```

Comment:
don't you mean `nullOr str`?

Reply:
Yes, I didn't realize null was coerced to empty string.